src/core/document.js

```diff
@@ -854,6 +856,71 @@ class PDFDocument {
     return this.xfaFactory !== null;
   }
 
+  async loadXfaFonts(handler, task) {
+    const acroForm = await this.pdfManager.ensureCatalog("acroForm");
+
+    const resources = await acroForm.getAsync("DR");
+    if (!(resources instanceof Dict)) {
+      return;
+    }
+    const objectLoader = new ObjectLoader(resources, ["Font"], this.xref);
+    await objectLoader.load();
+
+    const fontRes = resources.get("Font");
+    if (!(fontRes instanceof Dict)) {
+      return;
+    }
+
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: -1,
+      idFactory: this._globalIdFactory,
+      fontCache: this.catalog.fontCache,
+      builtInCMapCache: this.catalog.builtInCMapCache,
+    });
+    const operatorList = new OperatorList();
+    const initialState = {
+      font: null,
+      clone() {
+        return this;
+      },
+    };
+
+    const fonts = new Map();
+    fontRes.forEach((fontName, font) => {
+      fonts.set(fontName, font);
+    });
+    const promises = [];
+
+    for (const [fontName, font] of fonts) {
+      const descriptor = font.get("FontDescriptor");
+      if (descriptor instanceof Dict) {
+        const fontFamily = descriptor.get("FontFamily");
+        const fontWeight = descriptor.get("FontWeight");
+        const italicAngle = descriptor.get("ItalicAngle");
+        const cssFontInfo = { fontFamily, fontWeight, italicAngle };
+
+        if (!validateCSSFont(cssFontInfo)) {
+          break;
```

Comment:
Isn't `continue` what you want here, since otherwise *one* broken font could prevent all fonts from loading?

Reply:
Of course... Thx